openeogeotrellis/job_tracker_v2.py

```diff
@@ -68,6 +80,72 @@ def get_job_metadata(self, job_id: str, user_id: str, app_id: str) -> _JobMetada
         raise NotImplementedError
 
 
+class JobCostsCalculator(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def calculate_costs(self, job_info: dict, job_metadata: _JobMetadata, result_metadata: dict) -> float:
+        raise NotImplementedError
+
+
+class EtlApiJobCostsCalculator(JobCostsCalculator):
+    def __init__(self, etl_api: EtlApi, etl_api_access_token: str):
+        self._etl_api = etl_api
+        self._etl_api_access_token = etl_api_access_token
+
+    def calculate_costs(self, job_info: dict, job_metadata: _JobMetadata, result_metadata: dict) -> float:
```

Comment:
It feels there is some conceptual redundancy between `job_info` and `job_metadata`. 
Is is easily possible to eliminate one in favor of the other?

Reply:
Also possibly. This was very ad-hoc to get things started.